### ISSUE TITLE
Query using Fragment for Union gives empty array

### ISSUE THREAD

**Issue by MichaelDeBoey** [Youshido/GraphQL#151]
When using a Fragment when querying for a Union, you get an empty array
As you can see on [Launchpad](https://launchpad.graphql.com/kz7pmqrm7), this should work. 😕

schema:
```js
type Query {
  getAllPersons: PersonReturnType
}

type PersonReturnType {
  persons: [Person],
}
	
union Person = Instructor | Student

type Instructor {
  id: Int
  name: String
  salary: Float
}

type Student {
  id: Int
  name: String
  major: String
}
```

query:
```js
{
  getAllPersons {
    persons {
      ...personFrag
    }
  }
}

fragment personFrag on Person {
  __typename
  id
  name
  
  ... on Instructor {
    salary
  }
  
  ... on Student {
    major
  }
}
```

response:
```js
{
  "data": {
    "getAllPersons": {
      "persons": []
    }
  }
}
```

CC: @viniychuk

**Comment by fritz-gerneth** [comment]
For interfaces at least, this is working as expected (just checked it on our implementation). 

Looking at your query I actually first thought you're using interfaces as well (as you're querying for shared properties). Does this work as expected when moving all properties inside the unnamed fragment? Does this work when not using named fragments at all and only using unnamed fragments?

**Comment by MichaelDeBoey** [comment]
@fritz-gerneth If I don't use a fragment and just put the required fields inline, it works perfectly:
```js
{
  getAllPersons {
    persons  {
      __typename
      id
      name
      
      ... on Instructor {
        salary
      }
      
      ... on Student {
        major
      }
    }
  }
}
```

**Comment by viniychuk** [comment]
@MichaelDeBoey Thanks for reporting, will look into it. If you could create a test or just a all-in-one file that has the schema and request you're sending – that would help to expedite the solution... Thanks!

**Comment by MichaelDeBoey** [comment]
@viniychuk The whole schema is in [my first post](https://github.com/Youshido/GraphQL/issues/151#issue-233941362) or do you want some more info?

**Comment by viniychuk** [comment]
@MichaelDeBoey my question was about creating an actual PHP test / file we can use to test. While the creating of the test file is a really straightforward thing it helps a lot to get right to the bug and do it more quickly. Again, it's if you have time to do so, that would be much appreciated, if not – that's ok to, no worries.

**Comment by MichaelDeBoey** [comment]
@viniychuk I don't have that much time, but I'm happy to try if you can push me in the right direction?
The above schema + query is the one I used and can be used in a test, so it can't be that difficult to try and make a test for it I guess 🙂

**Comment by MichaelDeBoey** [comment]
@viniychuk @fritz-gerneth Any news on this one? 🙂

**Comment by portey** [comment]
@viniychuk fixed in branch "origin/issue-fixes"

**Comment by viniychuk** [comment]
released in 1.5.1
